services/brig/src/Brig/API/Internal.hs

```diff
@@ -171,6 +180,125 @@ authAPI =
+  db <- wrapClient Data.getFederationRemotes
+  (ms :: Maybe FederationStrategy, mf :: [FederationDomainConfig], mu :: Maybe Int) <- do
+    cfg <- ask
+    domcfgs <- remotesListFromCfgFile -- (it's not very elegant to prove the env twice here, but this code is transitory.)
+    pure
+      ( setFederationStrategy (cfg ^. settings),
+        domcfgs,
+        setFederationDomainConfigsUpdateFreq (cfg ^. settings)
+      )
+
+  -- update frequency settings of `<1` are interpreted as `1 second`.  only warn about this every now and
+  -- then, that'll be noise enough for the logs given the traffic on this end-point.
+  unless (maybe True (> 0) mu) $
+    randomRIO (0 :: Int, 1000)
+      >>= \case
+        0 -> Log.warn (Log.msg (Log.val "Invalid brig configuration: setFederationDomainConfigsUpdateFreq must be > 0, using default 10 seconds."))
+        _ -> pure ()
+
+  defFederationDomainConfigs
+    & maybe id (\v cfg -> cfg {strategy = v}) ms
+    & (\cfg -> cfg {remotes = nub $ db <> mf})
+    & maybe id (\v cfg -> cfg {updateInterval = min 1 v}) mu
+    & pure
+
+updateFederationRemotes :: Domain -> FederationDomainConfig -> ExceptT Brig.API.Error.Error (AppT r) ()
+updateFederationRemotes dom fedcfg = do
+  assertDomainIsNotUpdated dom fedcfg
+  assertNoDomainsFromConfigFiles dom
+  (lift . wrapClient . Data.updateFederationRemote $ fedcfg) >>= \case
+    True -> pure ()
+    False ->
+      throwError . fedError . FederationUnexpectedError . cs $
+        "federation domain does not exist and cannot be updated: " <> show (dom, fedcfg)
+
+assertDomainIsNotUpdated :: Domain -> FederationDomainConfig -> ExceptT Brig.API.Error.Error (AppT r) ()
+assertDomainIsNotUpdated dom fedcfg = do
+  when (dom /= domain fedcfg) $
+    throwError . fedError . FederationUnexpectedError . cs $
+      "federation domain of a given peer cannot be changed from " <> show (domain fedcfg) <> " to " <> show dom <> "."
+
+-- | FUTUREWORK: should go away in the future; see 'getFederationRemotes'.
+assertNoDomainsFromConfigFiles :: Domain -> ExceptT Brig.API.Error.Error (AppT r) ()
+assertNoDomainsFromConfigFiles dom = do
+  cfg <- asks (fromMaybe [] . setFederationDomainConfigs . view settings)
+  when (dom `elem` (domain <$> cfg)) $ do
+    throwError . fedError . FederationUnexpectedError $
+      "keeping track of remote domains in the brig config file is deprecated, but as long as we \
+      \do that, removing or updating items listed in the config file is not allowed."
+
+-- | Remove the entry from the database if present (or do nothing if not).  This responds with
+-- 533 if the entry was also present in the config file, but only *after* it has removed the
+-- entry from cassandra.
+deleteFederationRemotes :: Domain -> ExceptT Brig.API.Error.Error (AppT r) ()
+deleteFederationRemotes dom = do
+  lift . wrapClient . Data.deleteFederationRemote $ dom
+  assertNoDomainsFromConfigFiles dom
```

Comment:
The ordering on this delete then check seems weird, but allows us to default all the way back to config file state for a federation domain.

When we come to the point where we drop support for federation domains in the config files, we can remove most of these checks beyond "we know about this domain", and that any given domain isn't being updated to point at another domain. See `assertDomainIsNotUpdated`